### PR TITLE
KAFKA-8147: Add changelog topic configuration to KTable suppress

### PR DESCRIPTION
**Pull request by highluck**
KAFKA-8147: Add changelog topic configuration to KTable suppress

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by highluck** [comment]
@mjsax
Should I do it like this way?
thank you!

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -120,6 +120,11 @@ <h3><a id="streams_api_changes_260" href="#streams_api_changes_260">Streams API
         We added a <code>--force</code> option in StreamsResetter to force remove left-over members on broker side when long session time out was configured
         as per <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-571%3A+Add+option+to+force+remove+members+in+StreamsResetter">KIP-571</a>.
     </p>
+    <p>
+        We added a <code>Suppressed.withLoggingDisabled()</code> and <code>Suppressed.withLoggingEnabled(config)</code>
+        Suppressed API to disables logging and allows for configuration of the changelog topic
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: `Suppressed API` -> `method`

`to disable` (no `s`)
`and allowing`

File: docs/streams/upgrade-guide.html [hunk 2]
```diff
@@ -120,6 +120,11 @@ <h3><a id="streams_api_changes_260" href="#streams_api_changes_260">Streams API
...
+        as per <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-446%3A+Add+changelog+topic+configuration+to+KTable+suppress"></a>
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: `<a href="...">KIP-446</a>.`

There should be a link text :)

Also add `.` at the end of the sentence.

File: docs/streams/developer-guide/dsl-api.html [hunk 3]
```diff
@@ -3393,6 +3393,10 @@ <h5><a class="toc-backref" href="#id34">KTable-KTable Foreign-Key
 				      but this simple example creates a buffer with no
 				      upper bound.
 				    </dd>
+                    <dt><code>withLoggingDisabled()</code></dt>
+                    <dd>
+                      This configures the suppression operator to disable logging for changelog entries.
+                    </dd>
```
Comment (mjsax) [inline docs/streams/developer-guide/dsl-api.html]: Not sure if we want to add this in this example? \cc @vvcephei WDYT?

**Review by mjsax — COMMENTED** [review]
Thanks for the PR @highluck! Just a few minor comments.

**Comment by highluck** [comment]
@mjsax 
thank you for review!
updated code!

File: docs/streams/upgrade-guide.html [hunk 4]
```diff
@@ -120,6 +120,11 @@ <h3><a id="streams_api_changes_260" href="#streams_api_changes_260">Streams API
         We added a <code>--force</code> option in StreamsResetter to force remove left-over members on broker side when long session time out was configured
         as per <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-571%3A+Add+option+to+force+remove+members+in+StreamsResetter">KIP-571</a>.
     </p>
+    <p>
+        We added a <code>Suppressed.withLoggingDisabled()</code> and <code>Suppressed.withLoggingEnabled(config)</code>
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: just `We added <code>` (no `a`)

File: docs/streams/upgrade-guide.html [hunk 5]
```diff
@@ -120,6 +120,11 @@ <h3><a id="streams_api_changes_260" href="#streams_api_changes_260">Streams API
...
+        method to disables (no s) and allows for configuration of the changelog topic
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]: `method[s]` (plural)

`to disables (no s)...` -> `to allow disabling or configuring of the changelog topic`

**Review by mjsax — COMMENTED** [review]
Thanks for updating the PR @highluck

File: docs/streams/developer-guide/dsl-api.html (hunk 3 above)
Comment (vvcephei) [reply docs/streams/developer-guide/dsl-api.html]: Thanks @mjsax , I agree; this is an advanced configuration that shouldn't be used most of the time, so I wouldn't put it in the example.

Comment (highluck) [reply docs/streams/developer-guide/dsl-api.html]: update code thanks!

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.7` and `2.6` branches.

**Comment by mjsax** [comment]
Thanks a lot for the PR @highluck!

**Pull request merged**